### ISSUE TITLE
Keys may not preserve order

### ISSUE THREAD

**Issue by lua-rocks** [kithf/tomlua#2]
You noted in the README that keys may not preserve order when encoded. I don't really understand the point of this note since lua tables are also not preserve any order. Can you clarify this please?

**Comment by kithf** [comment]
When you deserialize a TOML document in Lua and then serialize it back into TOML, you can get a document with a different key order

Like this:
```lua
local doc, err = toml.decode[=[
title = "Example"
working = true
]=]

local src = toml.encode(doc)
print(src) 
--[[
working = true
title = "Example"
--]]
```

**Comment by lua-rocks** [comment]
Of course! Because you store it in the lua-table (the `doc` variable in your example) but lua-tables never keeps key order. This is not a TOML-related issue, but a basic lua-tables nature. TOML actually [should preserve it](https://github.com/toml-lang/toml/issues/162).

I think, `toml.decode` should return a third table where keys are order numbers and values are keys from `doc` to make possible to handle it correctly.
